### PR TITLE
feat: added iconMenu component

### PR DESCRIPTION
**Pull request by shayan-deriv**
Chages:
- added `IconMenu` component
- fixed some style issues for `MenuItem` component
- fixed minor bug in `PlatformSwitcher` component
- added support for overriding classname in `Tooltip` component

Screen Record:
https://github.com/deriv-com/ui/assets/100833613/6175bae5-299f-445f-903e-92f96f7485fd

### CODE REVIEW

File: src/components/Tooltip/index.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { ReactNode, useRef, useState } from "react";
+import React, { ComponentProps, ReactNode, useRef, useState } from "react";
```
Comment (niloofar-deriv) [inline src/components/Tooltip/index.tsx:1]:
```suggestion
import { ComponentProps, ReactNode, useRef, useState } from "react";
```

😁

**Comment by ali-hosseini-deriv** [comment]
dont we need a storybook story for it?

File: src/components/AppLayout/MenuItem/MenuItem.scss [hunk 2]
```diff
@@ -1,8 +1,13 @@
 .deriv-menu-item {
-    display: flex;
     align-items: center;
     height: 100%;
     cursor: pointer;
+    display: inline-flex;
+    justify-content: space-evenly;
+    padding: 0 16px;
```
Comment (niloofar-deriv) [inline src/components/AppLayout/MenuItem/MenuItem.scss:7]: Why you added these styles to this component?

**Comment by shayan-deriv** [comment]
> test cases are missing @shayan-deriv will appreciate if you write them as well.

I'm adding test and stories. I just wanted you guys to check the component implementation

File: src/components/AppLayout/IconMenu/index.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            <Tooltip
+                message={tooltipMesaage}
+                position={tooltipPosition}
+                className={clsx(
+                    "deriv-app-icon-menu",
+                    tooltipClassName,
+                )}
+                wrapperClassName={clsx("deriv-app-icon-menu__wrapper",className)}
+            >
+                <MenuItem
```
Comment (niloofar-deriv) [inline src/components/AppLayout/IconMenu/index.tsx:36]: I guess we need to add onclick or href as well

File: src/components/AppLayout/IconMenu/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { ComponentProps, ReactNode } from "react";
+import { Tooltip } from "../../Tooltip";
+import { MenuItem } from "../MenuItem";
+import "./IconMenu.scss";
+import clsx from "clsx";
+
+type IconMenuProps = ComponentProps<"div"> & {
+    isHoverDisabled?: boolean;
+    icon: ReactNode;
+    as: ComponentProps<typeof MenuItem>["as"];
```
Comment (niloofar-deriv) [inline src/components/AppLayout/IconMenu/index.tsx:10]:
```suggestion
```

File: src/components/AppLayout/IconMenu/index.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    wrapperClassName?: string;
+};
+export const IconMenu = ({
+    isHoverDisabled = false,
+    icon,
+    as,
+    tooltipMesaage,
+    tooltipClassName,
+    className,
+    tooltipPosition = "bottom",
```
Comment (niloofar-deriv) [inline src/components/AppLayout/IconMenu/index.tsx:23]:
```suggestion
    tooltipPosition = "bottom",
    ...menuItemProps
```

File: src/components/AppLayout/IconMenu/index.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                    "deriv-app-icon-menu",
+                    tooltipClassName,
+                )}
+                wrapperClassName={clsx("deriv-app-icon-menu__wrapper",className)}
+            >
+                <MenuItem
+                    className="deriv-app-icon-menu__item"
+                    disableHover={isHoverDisabled}
+                    leftComponent={icon}
+                    as={as}
```
Comment (niloofar-deriv) [inline src/components/AppLayout/IconMenu/index.tsx:40]:
```suggestion
                    {...menuItemProps}
```

**Pull request closed without merging**